src/lib/stitching/exchange/stitching.ts

```diff
@@ -0,0 +1,222 @@
+      resolve: (parent, _args, context, info) => {
+        const typename = parent[to].__typename
+        const id = parent[to].id
+
+        // Make a call to the user or partner resolver on query to
+        // grab our Metaphysics representations
+        return (
+          info.mergeInfo
+            .delegateToSchema({
+              schema: localSchema,
+              operation: "query",
+              fieldName: typename === "CommerceUser" ? "user" : "partner",
+              args: {
+                id,
+              },
+              context,
+              info,
+              transforms: (exchangeSchema as any).transforms,
+            })
+            // Re-jigger the type systems back into place, as right now
+            // it is considered a CommerceUser and clients will reject it.
+            .then(response => {
+              response.__typename =
+                typename === "CommerceUser" ? "User" : "Partner"
+              return response
+            })
+        )
+      },
+    }
+  }
+
+  const buyerDetailsResolver = partyUnionToDetailsFactory({
+    from: "buyer",
+    to: "buyerDetails",
+  })
+  const sellerDetailsResolver = partyUnionToDetailsFactory({
+    from: "seller",
+    to: "sellerDetails",
+  })
+  const fromDetailsResolver = partyUnionToDetailsFactory({
+    from: "from",
+    to: "fromDetails",
+  })
+
+  // Map the totals array to a set of resolvers that call the amount function
+  // the type param is only used for the fragment name
+  const totalsResolvers = (type, totalSDLS) =>
+    totalSDLS.map(name => ({
+      [name]: {
+        fragment: `fragment ${type}_${name} on ${type} { ${name}Cents }`,
+        resolve: (parent, args, _context, _info) =>
+          amount(_ => parent[name + "Cents"]).resolve({}, args),
+      },
+    }))
+
+  // Used to convert an array of `key: resolvers` to a single obj
+  const reduceToResolvers = arr => arr.reduce((a, b) => ({ ...a, ...b }))
+
+  return {
+    // The SDL used to declare how to stitch an object
+    extensionSchema: `
+    extend type CommerceLineItem {
+      artwork: Artwork
+      artworkVersion: ArtworkVersion
+      ${lineItemTotalsSDL.join("\n")}
+    }
+
+    extend type CommerceBuyOrder {
+      buyerDetails: OrderParty
+      sellerDetails: OrderParty
+
+      ${orderTotalsSDL.join("\n")}
+    }
+
+    extend type CommerceOfferOrder {
+      buyerDetails: OrderParty
+      sellerDetails: OrderParty
+
+      ${orderTotalsSDL.join("\n")}
+      ${amountSDL("offerTotal")}
+    }
+
+    extend interface CommerceOrder {
+      buyerDetails: OrderParty
+      sellerDetails: OrderParty
+
+      ${orderTotalsSDL.join("\n")}
+    }
+
+    extend type CommerceOffer {
+      fromDetails: OrderParty
+      ${offerAmountFieldsSDL.join("\n")}
+    }
+  `,
+
+    // Resolvers for the above
+    resolvers: {
+      CommerceBuyOrder: {
+        // The money helper resolvers
+        ...reduceToResolvers(totalsResolvers("CommerceBuyOrder", orderTotals)),
```

Comment:
Is it possible to do the `reduceToResolvers` part as part of `totalsResolvers` to reduce the noise a bit?